torch/utils/checkpoint.py

```diff
@@ -129,10 +129,15 @@ def checkpoint_sequential(functions, segments, *inputs):
     """
 
     def run_function(start, end, functions):
-        def forward(*inputs):
-            input = inputs[0]
+        def forward(input, *additional_inputs):
+            if not isinstance(input, tuple):
+                input = (input,)
```

Comment:
The special handling of the first argument to `forward` specifically seems a bit weird. Can you explain which case this makes work out?

Reply:
Agreed - the original intention was for the emptiness of `additional_inputs` to determine whether we had multiple arguments or not (i.e. whether we had multiple tensors as input), and then toggle function behavior based on that. I've cleaned up the code so that just one argument, inputs, needs to be passed in; we now determine whether we have multiple tensors as input and toggle based on that by looking at `isinstance(inputs, tuple)`.